app/lib/test_applications.rb

```diff
@@ -1,13 +1,22 @@
 module TestApplications
-  def self.create_application(states:)
+  def self.generate_for_provider(provider:, courses_per_application:, count:)
+    1.upto(count).flat_map do
+      create_application(
+        states: [:awaiting_provider_decision] * courses_per_application,
+        courses_to_apply_to: provider.courses,
```

Comment:
Should the list of courses be limited to courses that have course options? I got a crash later for a course that didn't have any - I was using Gorse SCCIT auto-synced which seems to have 5 out of 27 courses without options (my database may be messed up though).

Reply:
When I remove the courses with no options it works but I get 2200 new application choices. Not sure why - that's 100 * the number of courses.

Reply:
Aha — we're mapping over all the courses instead of only enough to generate applications in the relevant states 🤕